var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -38,67 +45,170 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
-    conflicts("+stdindices", when="%gcc@:10.1.0", msg="STD-indices requires newer version of GCC")
-    conflicts("+stdranges", when="%gcc@:10.1.0", msg="STD-ranges requires newer version of GCC")
+    conflicts("+std", when="%gcc@:10.1.0", msg="STD requires newer version of GCC")
 
     # CUDA conflict
     conflicts(
         "cuda_arch=none",
         when="+cuda",
         msg="CUDA requires architecture to be specfied by cuda_arch=",
     )
-    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
-    # Raja Conflict
     variant(
-        "offload", values=str, default="none", description="Enable RAJA Target [CPU or NVIDIA]"
-    )
-    conflicts(
-        "offload=none",
-        when="+raja",
-        msg="RAJA requires architecture to be specfied by acc_target=[CPU,NVIDIA]",
+        "cuda_memory_mode",
+        values=("default", "managed", "pagefault"),
+        default="default",
+        when="+cuda",
+        description="Enable MEM Target for CUDA",
     )
 
-    # download raja from https://github.com/LLNL/RAJA
+    # OMP offload
+    variant("omp_offload", default=False, when="+omp", description="Enable OpenMP Target")
+    variant(
+        "omp_flags",
+        values=str,
+        default="none",
+        when="+omp",
+        description="If OFFLOAD is enabled, this *overrides* the default offload flags",
+    )
     conflicts(
-        "dir=none",
+        "omp_flags=none",
+        when="+omp_offload",
+        msg="OpenMP requires offload flags to be specfied by omp_flags=",
+    )
+    # Raja offload
+    variant(
+        "raja_offload",
+        values=("cpu", "nvidia"),
+        default="cpu",
         when="+raja",
-        msg="RAJA implementation requires architecture to be specfied by dir=",
+        description="Enable RAJA Target [CPU or NVIDIA] / Offload with custom settings for OpenMP",
+    )
+    # std-* offload
+    variant(
+        "std_offload",
+        values=("nvhpc", "none"),
+        default="none",
+        when="+std",
+        description="Enable offloading support (via the non-standard `-stdpar`)\
+                    for the new NVHPC SDK",
+    )
+    variant(
+        "std_onedpl_backend",
+        values=("openmp", "tbb", "dpcpp", "none"),
+        default="none",
+        when="+std",
+        description="Implements policies using OpenMP,TBB or dpc++",
+    )
+    variant(
+        "std_use_tbb",
+        values=(True, False),
+        default=False,
+        when="+std",
+        description="No-op if ONE_TBB_DIR is set. Link against an in-tree oneTBB\
+                    via FetchContent_Declare, see top level CMakeLists.txt for details",
+    )
+    variant(
+        "std_use_onedpl",
+        values=(True, False),
+        default=False,
+        when="+std",
+        description="Link oneDPL which implements C++17 executor policies\
+                    (via execution_policy_tag) for different backends",
+    )
```

Comment:
The aim here was if the user provides `+std_use_onedpl` it will be enabled if not it will be disabled. Also, limits user from inputting `std_use_onedpl=[2024,OK] ` etc. and enables us not to use conflicts which was suggested in the earlier set of reviews actually.  The none method would be similar to this ?  (replacing false with none ?) 

https://github.com/spack/spack/blob/d70e9e131d12d9c1ee9f155ecb44684999d1f593/var/spack/repos/builtin/packages/itk/package.py#L42-L45